packages/task/src/browser/quick-open-task.ts

```diff
@@ -312,23 +319,39 @@ export class TaskAttachQuickOpenItem extends QuickOpenItem {
 }
 export class TaskConfigureQuickOpenItem extends QuickOpenGroupItem {
 
+    protected taskDefinitionRegistry: TaskDefinitionRegistry;
+
     constructor(
         protected readonly task: TaskConfiguration,
         protected readonly taskService: TaskService,
-        protected readonly labelProvider: LabelProvider
+        protected readonly labelProvider: LabelProvider,
+        protected readonly workspaceService: WorkspaceService,
+        protected readonly isMulti: boolean
     ) {
         super();
+        const stat = this.workspaceService.workspace;
+        this.isMulti = stat ? !stat.isDirectory : false;
```

Comment:
you should not cache such things, but always compute them from `WorkspaceService` to make sure that you read up-to-date state, please review others places in the PR

Generally try to have a single source for the some state and read it from there, don't replicate it. Replication sometimes helpful, usually between processes, but introduces with keeping replica in sync and invalidating it properly.

Reply:
`WorkspaceService` has `isMultiRootWorkspaceOpened`, just call it everywhere